packages/checkout/registry/index.ts

```diff
@@ -74,7 +74,7 @@ export const __experimentalApplyCheckoutFilter = < T >( {
 	/** Object containing arguments for the filter function. */
 	arg: CheckoutFilterArguments;
 	/** Function that needs to return true when the filtered value is passed in order for the filter to be applied. */
-	validation: ( value: unknown ) => true | Error;
+	validation?: ( value: T ) => true | Error;
```

Comment:
`value` type isn't uknown, we know for sure that it's the same as defaultValue and return value `T`